src/main/java/nextstep/subway/applicaion/GraphService.java

```diff
@@ -0,0 +1,38 @@
+package nextstep.subway.applicaion;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import nextstep.subway.applicaion.dto.PathResponse;
+import nextstep.subway.applicaion.dto.StationResponse;
+import nextstep.subway.domain.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class GraphService {
+    @Autowired
+    private LineRepository lineRepository;
+
+    @Autowired
+    private StationRepository stationRepository;
+
+    public PathResponse getShortestPath(Long source, Long target) {
+        List<Line> allLines = lineRepository.findAll();
+        Station sourceStation = stationRepository.findById(source).get();
+        Station targetStation = stationRepository.findById(target).get();
+
+        Graph graph = new Graph(allLines);
+
+        List<Station> path = graph.dijkstraShortestPath(sourceStation, targetStation);
+
+        int shortestDistance = graph.dijkstraShortestDistance(path);
+
+        List<StationResponse> stationResponses = path.stream().map(station ->
+                        new StationResponse(station))
+                .collect(Collectors.toList());
```

Comment:
stream API를 사용할 때 마다 개행을 해주면 stream pipeline을 파악하기가 더 편해집니다 :)

```suggestion
        List<StationResponse> stationResponses = path.stream()
                .map(StationResponse::new)
                .collect(Collectors.toList());
```